From 82221b4692562a7a7d9f82f0b79d714c96092cb8 Mon Sep 17 00:00:00 2001
From: Meriipu <Meriipu@users.noreply.github.com>
Date: Sat, 22 May 2021 15:35:27 +0200
Subject: [PATCH 04/10] skip the topsites provider due to horrible
 priority-stealing behaviour

even though history was getting sent to the query this provider seems
to somehow have dropped it and returned NOTHING because topsites are
disabled in my setup, leading to the dropmarker doing absolutely
nothing.
---
 browser/components/urlbar/UrlbarProvidersManager.jsm | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/browser/components/urlbar/UrlbarProvidersManager.jsm b/browser/components/urlbar/UrlbarProvidersManager.jsm
index 22ce3ce77012..c043b3817b29 100644
--- a/browser/components/urlbar/UrlbarProvidersManager.jsm
+++ b/browser/components/urlbar/UrlbarProvidersManager.jsm
@@ -369,6 +369,13 @@ class Query {
       //     return;
       //   }
       provider.queryInstance = this;
+
+      // Christ almighty
+      if (provider.name == "UrlbarProviderTopSites")
+      {
+        console.log("[UrlbarProvidersManager] Skipping provider: " + provider.name + " " + provider.type);
+        continue;
+      }
       activePromises.push(
         // Not all isActive implementations are async, so wrap the call in a
         // promise so we can be sure we can call `then` on it.  Note that
-- 
2.31.1

